packages/x-date-pickers/src/MonthCalendar/MonthCalendar.tsx

```diff
@@ -55,13 +56,15 @@ const MonthCalendarRoot = styled('div', {
   name: 'MuiMonthCalendar',
   slot: 'Root',
   overridesResolver: (props, styles) => styles.root,
-})<{ ownerState: MonthCalendarProps<any> }>({
+})<{ ownerState: MonthCalendarProps<any> }>(({ ownerState }) => ({
   display: 'flex',
   flexWrap: 'wrap',
   alignContent: 'stretch',
-  padding: '0 4px',
-  width: 320,
-});
+  padding: `0 ${ownerState.monthsPerRow === 3 ? 24 : 12}px`,
+  width: DIALOG_WIDTH,
+  // avoid padding increasing width over defined
+  boxSizing: 'border-box',
```

Comment:
Fixes the [same problem](https://github.com/mui/mui-x/pull/10135/files#diff-4017baaeb5c193d67ea78afd13f45961209b9f5c674be61af5d6b02ddcff7d7cR69) that was on `YearCalendar` as well 🙈